### ISSUE TITLE
Technical roadmap

### ISSUE THREAD

**Issue by jwheare** [ircv3/ircv3-specifications#241]
Edit: Roadmap milestone is now here: https://github.com/ircv3/ircv3-specifications/milestones/Roadmap

---

At the moment the IRC topic states:

> 3.3 bucket list: named modes, xinfo, login, alt to numerics, labeled replies, SRV records, strict transport security (#232), account registration and verification, basic JSON, roaming, SDP/ICE (like webrtc), etc.

It would be good to have a better indicator of what we're working on and a general order of priorities.

The easiest way is probably with github issue milestones, but we could also have a page on the website.

As far as I can tell, the main points of discussion lately have been:

* #242 STS
* #244 Metadata
* #236 Tags
* #246 Message ids, grouping and references

Some of those linked issues/PRs obsolete each other and can be consolidated, and some may make sense to be split off.

Some other PRs in the list that are almost mergeable but might still need some final disussion and decisions:

* cap-value batch for DEL/NEW updates: #168 
* REG/VERIFY for account creation/login: #152 
* <strike>STATUSMSG support for invite-notify: #206</strike>
* Server initial timestamping to manage client clock offsets: #199 (also #182)

The rest of the issues on the tracker are either not fleshed out ideas and can either be closed in future, moved to an ideas tracker repo or revisited.

Intents/CTCP/NOTICE replacement issues are tracked here: #247

The remaining items in the bucket list either fall into this category, never had a formal proposal suggested, or had one that was closed.

* xinfo #99 #28 (closed)
* roaming #76
* basic JSON #57
* SRV records #59 (closed)
* named modes
* alt to numerics
* SDP/ICE (like webrtc)

There has been discussion about JSON again recently so perhaps this should be properly revisited and opened up again. I'm not sure about the rest.

---

So in short, shall we create a new roadmap milestone (not tied to any version number like 3.3) containing the main items we've been discussing (plus JSON) as well as the PRs that are close to being ready, and then just link to it from IRC?

We can then do clean up on the remaining issues, move them to ircv3/ideas, or close them.

**Comment by jwheare** [comment]
Closing in favour of the milestone.
